### ISSUE TITLE
[3.3.5a] Glyph of Shadowflame

### ISSUE THREAD

**Issue by digz6666** [TrinityCore/TrinityCore#15610]
Glyph of Shadowflame is not working, is not applying 70% movement speed slow on every cast, it should be 100% chance. But it's not applying sometimes.
Seems like it absorbed, slow should be applied even though the damage absorbed.

Revision: 747a3f1d38abdb9b711877c168243f0d398b615f

![s1](https://cloud.githubusercontent.com/assets/1736110/10121647/1a414ff8-6529-11e5-942a-57e8aabcbda7.jpg)
![s2](https://cloud.githubusercontent.com/assets/1736110/10121648/1a426e10-6529-11e5-94cb-a5147117f185.jpg)

**Comment by P-Kito** [comment]
It has to do with absorb, I don't think a procChance is even existing for this

**Comment by digz6666** [comment]
@P-Kito I mean it has to apply 70% slow all the time, but in the screenshot you can see that no slow applied to attacker.

**Comment by cemak** [comment]
works for me:
```javascript
diff --git a/src/server/game/Spells/Auras/SpellAuras.cpp b/src/server/game/Spells/Auras/SpellAuras.cpp
index 4eea5ee..3be7c4f 100644
--- a/src/server/game/Spells/Auras/SpellAuras.cpp
+++ b/src/server/game/Spells/Auras/SpellAuras.cpp
@@ -1333,6 +1333,13 @@ void Aura::HandleAuraSpecificMods(AuraApplication const* aurApp, Unit* caster, b
                     if (target->HasAura(58039)) // Glyph of Blurred Speed
                         target->CastSpell(target, 61922, true); // Sprint (waterwalk)
                 break;
+            case SPELLFAMILY_WARLOCK:
+                if (!caster)
+                    break;
+                if (GetSpellInfo()->SpellIconID == 3317 && GetSpellInfo()->SpellVisual[0] == 11247) // Shadowflame
+                    if (caster->HasAura(63310)) // Glyph of Shadowflame
+                        caster->CastSpell(target, 63311, true);
+                break;
             case SPELLFAMILY_DEATHKNIGHT:
                 if (!caster)
                     break;

```

**Comment by Keader** [comment]
Fix  https://github.com/TrinityCore/TrinityCore/issues/15610#issuecomment-143539553 work for me too xD

**Comment by digz6666** [comment]
@cemak thank you for sharing your fix :+1:

**Comment by Demonid** [comment]
A lot of spells are not working correctly with absorb, for example Scourge Strike (Shadow part), Seals on hit damage etc

**Comment by digz6666** [comment]
@Demonid is there any issue covering that?

**Comment by Demonid** [comment]
@digz6666 Seems like Scourge Strike was already fixed some months ago https://github.com/TrinityCore/TrinityCore/pull/15035.

Seals i believe are still bugged: https://github.com/TrinityCore/TrinityCore/issues/11528

**Comment by P-Kito** [comment]
Scourge strike was not merged, this fix was not complete.

**Comment by Demonid** [comment]
I been using this code for almost 4 years and it "works", maybe this could help fix the issue?

https://gist.github.com/Demonid/b25382932be2e4d68124

**Comment by digz6666** [comment]
@Demonid Thank you for sharing your fix, will test and report :)

**Comment by digz6666** [comment]
@Demonid Thank you for sharing your fix, will test and report :)
By the way it needs changes damageInfo is undefined in Spell.cpp

```c++
@@ -2372,11 +2372,11 @@ void Spell::DoAllEffectOnTarget(TargetInfo* target)
         unitTarget->getHostileRefManager().threatAssist(caster, float(gain) * 0.5f, m_spellInfo);
         m_healing = gain;
 
         // Do triggers for unit (reflect triggers passed on hit phase for correct drop charge)
         if (canEffectTrigger && missInfo != SPELL_MISS_REFLECT)
-            caster->ProcDamageAndSpell(unitTarget, procAttacker, procVictim, procEx, addhealth, m_attackType, m_spellInfo, m_triggeredByAuraSpell);
+            caster->ProcDamageAndSpell(unitTarget, procAttacker, procVictim, procEx, damageInfo.damage + damageInfo.absorb, m_attackType, m_spellInfo, m_triggeredByAuraSpell);
     }
     // Do damage and triggers
     else if (m_damage > 0)
     {
         // Fill base damage struct (unitTarget - is real spell target)
@@ -2399,11 +2399,11 @@ void Spell::DoAllEffectOnTarget(TargetInfo* target)
             if (caster->GetTypeId() == TYPEID_PLAYER && m_spellInfo->HasAttribute(SPELL_ATTR0_STOP_ATTACK_TARGET) == 0 &&
                (m_spellInfo->DmgClass == SPELL_DAMAGE_CLASS_MELEE || m_spellInfo->DmgClass == SPELL_DAMAGE_CLASS_RANGED))
                 caster->ToPlayer()->CastItemCombatSpell(unitTarget, m_attackType, procVictim, procEx);
         }
 
-        m_damage = damageInfo.damage;
+        m_damage = damageInfo.damage + damageInfo.absorb;
 
         caster->DealSpellDamage(&damageInfo, true);
     }
     // Passive spell hits/misses or active spells only misses (only triggers)
     else
```

**Comment by Rushor** [comment]
@Demonid: the gist was deleted, pls readd it in a plaindiff in this issue.

**Comment by Aokromes** [comment]
If https://github.com/TrinityCore/TrinityCore/issues/15610#issuecomment-143539553 is valid, plz open a pr with it.

**Comment by ariel-** [comment]
The comment is a hack, this should be handled with the proc system.

**Comment by Demonid** [comment]
@ariel- Is this going to be fixed with your rework of the proc system ?

**Comment by cemak** [comment]
test it's:
`update spell_proc_event set procEx = procEx|0x0000400 where entry = 63310;`

**Comment by ariel-** [comment]
Fixed by e641d0c7d71bdb21e443bf52bf508b1581e07839 and 2ff855054f52bf2dcf81aa7a7da7bab7f8a99686
